### DISCUSSION TITLE
only half of the keyboard changes waveform / virtual split point

### DISCUSSION THREAD

**Discussion by rezirezi** [marcel-licence/esp32_basic_synth#102] [Q&A] [answered]
Hi,
First of all, I wanna say thanks for the amazing project you've made and the hard work you put into it. 
Now to the issue. 
So I added a button directly to the esp32 to toggle through the waveforms, It works but only the upper half of the notes change the waveform, the rest remain unchanged. 
On a button press I am manually calling the Synth_SetParam(8, float value) function to change the waveform, am I doing something wrong/ am I missing something?
I have tried changing both waveforms, together and separately too but no luck.

Thank you

**Comment by marcel-licence** [comment, marked as answer]
Hi,
thank you very much.
I think the virtual split point might be active. This allows to play two sounds via different channels and with different settings at the same time.

In z_config.ino you will find the redirection to the MidiCtrl which includes the split point as-well:
```
#ifdef MIDI_CTRL_ENABLED
    MidiCtrl_NoteOn,
    MidiCtrl_NoteOff,
#else
```

Hope this helps

**Reply by rezirezi** [reply]
Thanks!!
